rules/rules_test.go

```diff
@@ -667,6 +673,124 @@ func TestExpressions(t *testing.T) {
 			},
 		},
 		{
+			expr: `http_requests{group="canary"} and http_requests{instance="0"}`,
+			output: []string{
+				`http_requests{group="canary", instance="0", job="api-server"} => 300 @[%v]`,
+				`http_requests{group="canary", instance="0", job="app-server"} => 700 @[%v]`,
+			},
+		},
+		{
+			expr: `(http_requests{group="canary"}+1) and http_requests{instance="0"}`,
```

Comment:
That would be many-to-one matching, though. Sure `on(instance,job)` is what you have to do - should that be working different?